src/bundlers/kotlin/kotlinMain.ts

```diff
@@ -0,0 +1,135 @@
+import org.http4k.core.then
+import org.http4k.filter.AllowAll
+import org.http4k.filter.CorsPolicy
+import org.http4k.filter.OriginPolicy
+import org.http4k.filter.ServerFilters
+import org.http4k.routing.bind
+import org.http4k.routing.routes
+import org.http4k.server.SunHttp
+import org.http4k.server.asServer
+import kotlinx.serialization.json.jsonArray
+import com.amazonaws.services.lambda.runtime.Context
+import com.amazonaws.services.lambda.runtime.RequestHandler
+
+@Serializable
+data class JsonRpcRequest(
+        val jsonrpc: String,
+        val id: Int,
+        val method: String,
+        val params: List<JsonElement>?,
+)
+
+@Serializable
+data class JsonRpcResponse(val jsonrpc: String, val id: Int, val result: String)
+
+class LambdaHandler : RequestHandler<Map<String, Any>, String> { 
+  override fun handleRequest(event: Map<String, Any>, context : Context): String {
+      println(event)
+      return processRequest(event)
+  }
+}
+
+fun processRequestDirect(json_rpc : String): String {
+  return processRequest(mapOf("body" to json_rpc))
+}
+
+fun processRequest(event: Map<String, Any>): String {
+    println("PROCESSING REQ")
+    // Get request body
+    var request: JsonRpcRequest = JsonRpcRequest("", 2, "", listOf())
+    var req_id = 0
+    try {
+        var eventBody = event["body"].toString().replace("\\\\", "")
+        if(eventBody.get(0) != '"') {
+          eventBody = "\\"" + eventBody + "\\""
+        }
+        request = Json.decodeFromString(eventBody.substring(1, eventBody.length - 1))
+        req_id = request.id
+    } catch (e: Exception) {
+        println(e)
+        return "{\\"jsonrpc\\":\\"2.0\\",\\"result\\":\\"There was a problem with the request format.\\",\\"id\\":\\"$req_id\\"}"
+    }
+
+
+    val method = request.method
+    val params = request.params
+    
+    // Test if class is instantiable
+    try {
+        {{className}}()
+        println("Class instantiable")
+    } catch (e: Exception) {
+        println(e)
+        return "{\\"jsonrpc\\":\\"2.0\\",\\"result\\":\\"Class could not be instantiated. Check logs for more information.\\",\\"id\\":\\"$req_id\\"}"
+    }
+
+    when (method) {
+        {{#jsonRpcMethods}}
+        "{{className}}.{{name}}" -> {
+            {{#parameters}}
+            var param{{index}} = {{{cast}}}
+            {{/parameters}}
+            var func_res = {{className}}().{{name}}({{#parameters}}param{{index}}{{^last}},{{/last}}{{/parameters}});
+            var res_str = Json.encodeToString(func_res)
+            return "{\\"jsonrpc\\":\\"2.0\\",\\"result\\":$res_str,\\"id\\":\\"$req_id\\"}"
+        }
+        {{/jsonRpcMethods}}
+        else -> {
+            println("method not found")
+            return "{\\"jsonrpc\\":\\"2.0\\",\\"result\\":\\"Method not found.\\",\\"id\\":\\"$req_id\\"}"
+        }
+    }
+}
+
+fun main(args: Array<String>) {
+  var port: Int = 8080
+  if (args.size == 1) {
+    port = Integer.parseInt(args[0])
+  }
```

Comment:
This is going to be deployed on a Function (AWS Lambda or Genezio Runtime), how can I access this command line argument to change the port?

Reply:
This is only used for local deployments, the code has been split so there will be no more confusions